neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -0,0 +1,422 @@
+                            ValidatorState state_validator = ValidatorState.FromByteArray(storage_validator.Value);
+                            state_validator.Votes -= amount;
+                            storage_validator.Value = state_validator.ToByteArray();
+                        }
+                        StorageItem storage_count = engine.Service.Snapshot.Storages.GetAndChange(CreateStorageKey(ScriptHash, Prefix_ValidatorsCount));
+                        ValidatorsCountState state_count = ValidatorsCountState.FromByteArray(storage_count.Value);
+                        state_count.Votes[state_from.Votes.Length - 1] -= amount;
+                        storage_count.Value = state_count.ToByteArray();
+                    }
+                    StorageKey key_to = CreateStorageKey(ScriptHash, Prefix_Account, to);
+                    StorageItem storage_to = engine.Service.Snapshot.Storages.GetAndChange(key_to, () => new StorageItem
+                    {
+                        Value = new AccountState().ToByteArray()
+                    });
+                    AccountState state_to = AccountState.FromByteArray(storage_to.Value);
+                    DistributeGas(engine, to, state_to);
+                    state_to.Balance += amount;
+                    storage_to.Value = state_to.ToByteArray();
+                    if (state_to.Votes.Length > 0)
+                    {
+                        foreach (ECPoint pubkey in state_to.Votes)
+                        {
+                            StorageItem storage_validator = engine.Service.Snapshot.Storages.GetAndChange(CreateStorageKey(ScriptHash, Prefix_Validator, pubkey.ToArray()));
+                            ValidatorState state_validator = ValidatorState.FromByteArray(storage_validator.Value);
+                            state_validator.Votes += amount;
+                            storage_validator.Value = state_validator.ToByteArray();
+                        }
+                        StorageItem storage_count = engine.Service.Snapshot.Storages.GetAndChange(CreateStorageKey(ScriptHash, Prefix_ValidatorsCount));
+                        ValidatorsCountState state_count = ValidatorsCountState.FromByteArray(storage_count.Value);
+                        state_count.Votes[state_to.Votes.Length - 1] += amount;
+                        storage_count.Value = state_count.ToByteArray();
+                    }
+                }
+            }
+            engine.Service.SendNotification(engine, ScriptHash, new StackItem[] { "Transfer", from, to, amount });
+            return true;
+        }
+
+        private static void DistributeGas(ApplicationEngine engine, byte[] account, AccountState state)
+        {
+            BigInteger gas = CalculateBonus(engine, state.Balance, state.BalanceHeight, engine.Service.Snapshot.PersistingBlock.Index);
+            state.BalanceHeight = engine.Service.Snapshot.PersistingBlock.Index;
```

Comment:
Maybe we can check the height, and if is less than X blocks, return false?

Reply:
What are the benefits of this?

Reply:
We will be distribute gas in every transfer?

Reply:
We should try to prevent the spam of claim gas

Reply:
Now users needn't claim gas. They will get gas after transfer neo.

Reply:
Until reach this part of the code I was thinking about that.
@erikzhang, this is really the improvement we were waiting for. Congrulations.